### PR TITLE
Document aws IID attestation

### PR DESCRIPTION
**Pull request by walmav**
Provides documentation for configuring AWS IID based attestation

### CODE REVIEW

File: doc/aws/Setting_Up_IID_Attestation.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+### Installing SPIRE Server/Agent:
+
+Both SPIRE Server and SPIRE Agent binaries are bundled in the same distribution and can be found [ here ](https://github.com/spiffe/spire/releases/latest).
+Download and extract the SPIRE distribution into ~/opt directory:
+
+        mkdir ~/opt
+        wget https://github.com/spiffe/spire/releases/download/<release_version>/spire-<release_version>-linux-x86_64-glibc.tar.gz
+        tar -C ~/opt -xzf spire-<release_version>-linux-x86_64-glibc.tar.gz
+
+### Installing aws IID attestor plugin on SPIRE server:
```
Comment (ajessup) [inline doc/aws/Setting_Up_IID_Attestation.md]: This is really "building" as well as installing. Suggest breaking it our accordingly.

File: doc/aws/Setting_Up_IID_Attestation.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+1.  Download and extract go binary into /usr/local:
```
Comment (ajessup) [inline doc/aws/Setting_Up_IID_Attestation.md]: It's probably OK to just call out Go and Git as dependencies, and leave installing them as an exercise for the reader.

File: doc/aws/Setting_Up_IID_Attestation.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+### Configuring SPIRE Server plugin:
+
+Remove join-token attestor config and create AWS IID attestor config:
+
+```
+rm ~/opt/spire-<version>/conf/server/plugin/join-token.conf
+echo 'pluginName = "aws_iid_attestor"
+pluginCmd = "~/go/bin/server"
```
Comment (ajessup) [inline doc/aws/Setting_Up_IID_Attestation.md]: Where do I found out what all these configuration options actually do? How many of them are specific to the plugin?

File: doc/aws/Setting_Up_IID_Attestation.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+
+### Installing aws IID attestor plugin on SPIRE agent:
+
+Follow the same steps 1 thru 4 from Installing aws IID attestor plugin on SPIRE server above.
+
+5. Install agent IID attestor plugin using go install. The plugin binaries are installed in $GOBIN directory
+
+```
+go install github.com/spiffe/aws-iid-attestor/agent
```
Comment (ajessup) [inline doc/aws/Setting_Up_IID_Attestation.md]: Is there a way to install this without using Go? What if I built a binary on another machine using the instructions above?

File: doc/aws/Setting_Up_IID_Attestation.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+pluginData {
+    access_id = "<aws_access_key_id>"
+    secret = "<aws_access_secret_key>"
+    trust_domain = "example.org"
+}' > ~/opt/spire-<version>/conf/server/plugin/aws-iid-attestor.conf
+```
+
+### Configure and start SPIRE server
+
+Set `BindAddress= "<internal-ip>"` in ~/opt/spire-<version>/conf/server/server.conf to the internal/private IP of the server.
```
Comment (ajessup) [inline doc/aws/Setting_Up_IID_Attestation.md]: This feels like something that could be delegated to a more general tutorial.

**Comment by walmav** [comment]
Closing this and moved the documentation in the aws-iid-attestor repo: https://github.com/spiffe/aws-iid-attestor/pull/5

**Pull request closed without merging**